src/pages/nav-box.js

```diff
@@ -101,6 +121,124 @@ class NavBox extends HTMLElement {
       }
     }
   }
+
+  // Qr-code State Management
+
+  hideQrCodePopup() {
+    if (this._qrPopup) {
+      this._qrPopup.classList.remove("open");
+      this._qrPopup.classList.add("close");
+      setTimeout(() => {
+        this._qrPopup.remove();
+        this._qrPopup = null;
+      }, 300);
+    }
+  }
+
+  _toggleQrCodePopup() {
+    if (this._qrPopup) {
+      this.hideQrCodePopup();
+      return;
+    }
+
+    const currentUrl = this.querySelector("#url").value;
+    if (!currentUrl) return;
+
+    this._qrPopup = document.createElement("div");
+    this._qrPopup.className = "qr-popup";
+    this._qrPopup.innerHTML = `
+    <div class="qr-popup-header">
+      <p>Scan QR Code</p>
+      <button class="close-btn">
+        <svg width="18" height="18" fill="currentColor" viewBox="0 0 24 24">
+          <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/>
+        </svg>
+      </button>
+    </div>
+    <qr-code src="${currentUrl}" data-bg="white" data-fg="black"></qr-code>
+    <span class="qr-url">${currentUrl}</span>
+    <button class="download-btn disabled">Download</button>
+  `;
```

Comment:
Let's have a separate web component for this in `static/js/qr-popup.js`.

Also, we should use the svg for close button from `/static/assets/svg/close.svg`.